lib/cdo/contact_rollups.rb

```diff
@@ -337,6 +352,17 @@ def self.update_roles
     log "Updating Regional Partner role"
     append_regional_partner_to_role_list
     log_completion(start)
+
+    start = Time.now
+    log "Updating CSF/CSD/CSP teacher roles"
+    # CSF scripts don't have a course mapping - identify CSF teachers by
+    # specific scripts
+    add_role_from_script_sections_taught("CSF Teacher",CSF_SCRIPT_LIST)
+    # CSD and CSP scripts are mapped to course - identify CSD/CSP teachers
+    # that way
+    add_role_from_course_sections_taught("CSD Teacher","csd")
+    add_role_from_course_sections_taught("CSP Teacher","csp")
+    log_completion(start)
```

Comment:
nit: Not necessary for this PR, but I see this pattern a lot here:
```ruby
start = Time.now
log "Doing stuff"
# do stuff...
log_completion(start)
```

I think it would be a lot nicer, not to mention less likely for future errors, if this was a block - something like:

```ruby
def log_action(name)
  log name
  start = Time.now
  yield
  log_completion(start)
end
```

Then all the usages can be simplified to:
```ruby
log_action "Doing stuff" do
  # Do stuff...
end
```

Reply:
Thanks, good idea, will probably do that in the future.